src/native/fs/fs_napi.cpp

```diff
@@ -1941,6 +1948,51 @@ struct DirOpen : public FSWorker
     }
 };
 
+struct TellDir : public FSWrapWorker<DirWrap>
+{
+    long int _tell_res = -1;
+    TellDir(const Napi::CallbackInfo& info)
+        : FSWrapWorker<DirWrap>(info)
+    {
+        Begin(XSTR() << "TellDir " << DVAL(_wrap->_path));
+    }
+    virtual void Work()
+    {
+        DIR* dir = _wrap->_dir;
+        DBG1("FS::Telldir::Work: " << DVAL(_wrap->_path));
+        _tell_res = telldir(dir);
+        if (_tell_res == -1) SetSyscallError();
+    }
+    virtual void OnOK()
+    {
+        DBG0("FS::Telldir::OnOK: " << DVAL(_wrap->_path) << DVAL(_tell_res));
+        Napi::Env env = Env();
+        auto res = Napi::BigInt::New(env, _tell_res);
+        _deferred.Resolve(res);
+    }
+};
+
+struct SeekDir : public FSWrapWorker<DirWrap>
+{
+    long int _seek_pos;
+    SeekDir(const Napi::CallbackInfo& info)
+        : FSWrapWorker<DirWrap>(info)
+    {
+        bool lossless = true;
+        _seek_pos = info[1].As<Napi::BigInt>().Int64Value(&lossless);
+        if (!lossless) {
+            SetError("seekdir bigint value must convert lossless");
+        }
+        Begin(XSTR() << "SeekDir " << DVAL(_wrap->_path) << DVAL(_seek_pos));
+    }
+    virtual void Work()
+    {
+        DIR* dir = _wrap->_dir;
```

Comment:
add a check like we do in DirReadEntry for the case that dir was closed - if (!dir) ...